### PR TITLE
feat: add badge when waiting for signature on multisig transactions

### PR DESCRIPTION
**Pull request by davimello28**
Display a little badge inside the multisignature menu tab when waiting for that wallet signature.

image

## Summary

Since the logic inside fetching the waiting transactions resides inside the UI component, the store is changed according to its transactions. The correct way should be either refactoring the component by creating a service for the transactions or moving the whole logic to the store. 

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests
- [x] Ready to be merged

### CODE REVIEW

File: src/renderer/store/modules/transaction.js [hunk 1]
```diff
@@ -287,6 +292,14 @@ export default {
         networkId: rootGetters['session/profile'].networkId,
         staticFees
       })
+    },
+
+    /**
+     * Sets if there is a multisignature ttransaction waiting for this wallet signature.
```
Comment (brenopolanski) [inline src/renderer/store/modules/transaction.js]: @davimello28 change from `ttransaction` to `transaction`.

File: src/renderer/components/Wallet/WalletTransactions/WalletTransactionsMultiSignature.js [hunk 2]
```diff
@@ -65,6 +65,11 @@ export default {
         if (publicKey === WalletService.getPublicKeyFromWallet(this.wallet_fromRoute)) {
           this.$set(this, 'fetchedTransactions', response.transactions)
           this.totalCount = response.totalCount
+          if (this.totalCount > 0) {
```
Comment (alexbarnsley) [inline src/renderer/components/Wallet/WalletTransactions/WalletTransactionsMultiSignature.js]: I think the check needs improving here to only include transactions that require the wallet's signature at that point in time. I would do something like this:

```
let hasPendingTransaction = false
for (const transaction of response.transactions) {
  if (TransactionService.needsWalletSignature(transaction, WalletService.getPublicKeyFromWallet(this.wallet_fromRoute))) {
    hasPendingTransaction = true
    break
  }
}

if (hasPendingTransaction) {
  this.$store.dispatch('transaction/setWaitingMultisignature', true)
} else {
  this.$store.dispatch('transaction/setWaitingMultisignature', false)
}
```

This is used in the TransactionShowMultiSignature.vue component

Comment (davimello28) [reply src/renderer/components/Wallet/WalletTransactions/WalletTransactionsMultiSignature.js]: Alex, I did some little improvements. This is a computed property. 

```js
isWaitingForSignature () {
  const transactions = this.fetchedTransactions
  const wallet = WalletService.getPublicKeyFromWallet(this.wallet_fromRoute)

  function checkTransactionForWaitingSignatures (transaction) {
    return TransactionService.needsWalletSignature(
      transaction,
      wallet
    )
  }

  return transactions.some(t => checkTransactionForWaitingSignatures(t))
}
```

**Comment by faustbrian** [comment]
Closing due to other changes that are coming which nullify this.

**Pull request closed without merging**